aarch64_extension.cpp

```diff
@@ -275,6 +328,108 @@ class AArch64ArchitectureExtension : public ArchitectureHook {
     return false;
   }
 
+  bool LiftFMOV(cs_insn* instr, LowLevelILFunction& il) {
+    // 00003edc
+    cs_arm64* detail = &(instr->detail->arm64);
+
+    if (detail->op_count != 2) {
+      return false;
+    }
+
+    if (detail->operands[0].type != ARM64_OP_REG
+        || detail->operands[1].type != ARM64_OP_REG) {
+      return false;
+    }
+
+    uint32_t Rd = this->m_base->GetRegisterByName(
+        cs_reg_name(mCapstone, detail->operands[0].reg));
+    uint32_t Rn = this->m_base->GetRegisterByName(
+        cs_reg_name(mCapstone, detail->operands[1].reg));
+
+    size_t Rd_size = this->m_base->GetRegisterInfo(Rd).size;
+    size_t Rn_size = this->m_base->GetRegisterInfo(Rd).size;
+
+    if (Rd_size != Rn_size) {
+      return false;
+    }
+
+    il.AddInstruction(
+        il.SetRegister(Rd_size, Rd, il.Register(Rd_size, Rn)));
+
+    return true;
+  }
+
+  bool LiftMRS(cs_insn* instr, LowLevelILFunction& il) {
+    // 00003edc
+    cs_arm64* detail = &(instr->detail->arm64);
+
+    if (detail->op_count != 2) {
+      return false;
+    }
+
+    if (detail->operands[0].type != ARM64_OP_REG
+        || detail->operands[1].type != ARM64_OP_REG_MRS) {
+      return false;
+    }
+
+    uint32_t Rd = this->m_base->GetRegisterByName(
+        cs_reg_name(mCapstone, detail->operands[0].reg));
+    uint32_t Rn = detail->operands[1].reg;
+
+    size_t Rd_size = this->m_base->GetRegisterInfo(Rd).size;
+    size_t Rn_size = 8;
+
+    if (Rd_size != Rn_size) {
+      return false;
+    }
+
+    il.AddInstruction(
+        il.Intrinsic(vector<RegisterOrFlag> { RegisterOrFlag(false, Rd) },
+                     mIntrinsicNames["__builtin_mrs"],
+                     vector<ExprId> { il.Const(Rn_size, Rn, ILSourceLocation(instr->address, 1)) },
```

Comment:
Tying the intrinsic to Capstone constants is probably not a good idea... do we have any other options or will this have to do?

Reply:
It's at least not a meaningless constant in capstone, it's the operand parts of the instruction encoding, so you can look it up in the manual.
I'd prefer if we could actually show the register names, but without defining a pair of intrinsics for every register I'm not sure how to do that in binja's IL.

Reply:
Ok, this'll be good for now and we can make an issue to fix it later. I suspect we can write some custom display for the intrinsic to allow us to show the actual name.